Test/SqlDom/TSqlParserTest.cs

```diff
@@ -460,6 +460,57 @@ UPDATE dbo.Departments
                      return procedureStatementBody.StatementList.StartLine.Equals(5) && procedureStatementBody.StatementList.StartColumn.Equals(1);
                  }
             ));
+        }
+
+        [TestMethod]
+        [Priority(0)]
+        [Timeout(GlobalConstants.DefaultTestTimeout)]
+        [SqlStudioTestCategory(Category.UnitTest)]
+        public void ParsingSelectWithOptimizerHint()
+        {
+            string input = "";
+            IList<ParseError> errors = null;
+            TSqlScript script = null;
+
+            input = @"Select 1 option(
+HASH  GROUP, 
+ORDER  GROUP, 
+CONCAT UNION, 
+HASH UNION,
+MERGE UNION,
+Keep UNION,
+LOOP  JOIN, 
+HASH  JOIN, 
+MERGE  JOIN,  
+PARAMETERIZATION Simple,
+IGNORE_NONCLUSTERED_COLUMNSTORE_INDEX, 
+KEEP PLAN, 
+KEEPFIXED PLAN,
+recompile, 
+EXPAND VIEWS)";
+            errors = null;
+            script = (TSqlScript)new TSql160Parser(true).Parse(new StringReader(input), out errors);
+            Assert.IsTrue(errors.Count == 0, "Unexpected parsing error");
+
+            var selectStatement = (SelectStatement) script.Batches[0].Statements[0];
+            Assert.IsNotNull(selectStatement);
+            Assert.IsTrue(selectStatement.OptimizerHints.Any());
+
+            SqlScriptGenerator scriptGenerator = new Sql160ScriptGenerator();
+            scriptGenerator.GenerateScript(script, out string actualResult);
+            Assert.IsTrue(actualResult.IndexOf("option", StringComparison.OrdinalIgnoreCase) > 0);
+
+            foreach (var hint in selectStatement.OptimizerHints)
+            {
+                Assert.IsTrue(hint.FirstTokenIndex > 0 && hint.LastTokenIndex > 0);
```

Comment:
Add an error message, maybe something like `Assert.IsTrue(..., $"Invalid token indexes for {hint}");`